docs: explain rootless limitation and constraint (#5809)
dagger/dagger: core/docs/d7yxc-operator_manual.md

1. If using a pre-installed CLI, install the CLI at that version as detailed in [Versioning](#versioning).
1. Once the runner and/or CLI are setup, you are safe to upgrade your SDK to the newest version.

### Can I run the Dagger Engine as a "rootless" container?

No. "Rootless mode" means running the Dagger Engine as a container without the `--privileged` flag. In this case, the container would not have "root" privileges for all devices on the host system. Currently, the Dagger Engine cannot be run as a rootless container; network, platform and volume constraints related to rootless usage would significantly limit its capabilities and performance.

> #### Platform constraints
> 
> The Dagger Engine uses Buildkit. In order to run it in rootless mode, it is necessary to use [Rootlesskit](https://github.com/rootless-containers/rootlesskit), an experimental tool that aims to replace the "root" user with a "fake root" user using user namespaces. However, depending on the host operating system, some features might not be fully supported.Since Dagger is intended to run consistently everywhere, introducing platform-specific incompatibilities would be contrary to this goal.

Reviewer: ```suggestion
```

I think this section is mostly redundant with the below ones since the other sections talk about the various platform-related problems we run into w/ rootless containers.

If we do prefer to leave this section in, we don't need to mention rootlesskit specifically since there's many ways of approaching it. We can just say "it is necessary to use user namespaces".

> #### Volume constraints

Reviewer: ```suggestion
#### Filesystem constraints
```

> The Dagger Engine relies on the `overlayfs` snapshotter for efficient operation of mounted filesystems. In order to run it in rootless mode, it is necessary to use [`fuse-overlayfs`](https://github.com/containers/fuse-overlayfs), a rootless implementation of `overlayfs`. However, optimal configuration of `fuse-overlayfs` depends on the kernel capabilities, its own configuration, `fuse` availability and upstream kernel patches specific to the host distribution. In the absence of all these, correctly configured, you will experience noticeable slowdown, either with `fuse-overlayfs` or the default snapshotter. Since Dagger aims to accelerate your CI/CD pipeline, reducing performance also reduces the value provided by Dagger and leads to unstable behaviour.

Reviewer: ```suggestion
The Dagger Engine relies on the `overlayfs` snapshotter for efficient construction of container filesystems. However, only relatively recent Linux kernel versions fully support `overlayfs` inside of rootless user namespaces. On older kernels, there are fallback options such as [`fuse-overlayfs`](https://github.com/containers/fuse-overlayfs), but they come with their own complications in terms of degraded performance and host-specific setup.

We've not yet invested in the significant work it would take to support+document running optimally on each kernel version, hence the limitation at this time.
```

Clarified a few details since technically `overlayfs` is possible to use on new enough kernels. Also trimmed some of the detail about fuse-overlayfs since I'm not sure it's relevant to readers at that level of detail.

#### Network constraints

> Running the Dagger Engine in rootless mode constrains network management. It is possible to use [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but this tool significantly decreases network performance in addition to being extremely complex to setup. This [comparison table of network drivers](https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers) shows that `slirp` is at least five time slower than a root-privileged network driver.

Reviewer 2: ```suggestion
Running the Dagger Engine in rootless mode constrains network management. It is possible to use [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but this tool significantly decreases network performance in addition to being extremely complex to setup. This [comparison table of network drivers](https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers) shows that `slirp` is at least five times slower than a root-privileged network driver.
```

> Running the Dagger Engine in rootless mode constrains network management. It is possible to use [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but this tool significantly decreases network performance in addition to being extremely complex to setup. This [comparison table of network drivers](https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers) shows that `slirp` is at least five times slower than a root-privileged network driver.

Reviewer: ```suggestion
Running the Dagger Engine in rootless mode constrains network management due to the fact that it's not possible for a rootless container to move a network device from the host network namespace to its own network namespace. 

It is possible to use userspace TCP/IP implementations such as [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but they often significantly decrease network performance. This [comparison table of network drivers](https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers) shows that `slirp` is at least five times slower than a root-privileged network driver.

Newer options for more performant userspace network stacks have arisen in recent years, but they are generally either reliant on relatively recent kernel versions or in a nascent stage that would require significant validation around robustness+security.
```

Added a little more detail on the specific constraint we hit. Also clarified that `slirp` is only one option.

(11-12) `dagger session` listens on a random available port on localhost and generates a random uuid to use as a session token (same as described in the [previous section](#dsi-basic)). The port being listened on and the session token are serialized to JSON and then written to stdout, which is a pipe connected back to the SDK parent process.

(13-17) The SDK then sends GraphQL HTTP requests to the localhost listener, including the session token as a basic auth header.

> ### Why we do not support rootless engine?

Reviewer: I'm not sure we need this separate section, I think the first one is enough.

Author: Yeah It's pretty much the same, I'll remove it
